pkg/analysis/l2/l2analyzer.go

```diff
@@ -0,0 +1,96 @@
+package l2
+
+import (
+	"bytes"
+	"encoding/json"
+	"io"
+	"net/http"
+	"nodemon/pkg/entities"
+	"strconv"
+	"time"
+
+	"go.uber.org/zap"
+)
+
+type Response struct {
+	Jsonrpc string `json:"jsonrpc"`
+	ID      string `json:"id"`
+	Result  string `json:"result"`
+}
+
+func hexStringToInt(hexString string) (int64, error) {
+	// Parse the hexadecimal string to integer
+	return strconv.ParseInt(hexString[2:], 16, 64)
+}
+
+func collectL2Height(url string, ch chan<- int64, logger *zap.Logger) {
+	requestBody, err := json.Marshal(map[string]interface{}{
+		"jsonrpc": "2.0",
+		"id":      "1",
+		"method":  "eth_blockNumber",
+		"params":  []interface{}{},
+	})
+	if err != nil {
+		logger.Error("Failed to build a request body for l2 node", zap.Error(err))
+	}
+
+	resp, err := http.Post(url, "application/json", bytes.NewBuffer(requestBody))
+	if err != nil {
+		logger.Error("Failed to send a request to l2 node", zap.Error(err))
+		return
+	}
+	defer resp.Body.Close()
+
+	body, err := io.ReadAll(resp.Body)
+	if err != nil {
+		logger.Error("Failed to read response body from l2 node", zap.Error(err))
+		return
+	}
+
+	var response Response
+	err = json.Unmarshal(body, &response)
+	if err != nil {
+		logger.Error("Failed unmarshalling response:", zap.Error(err))
+		return
+	}
+
+	height, err := hexStringToInt(response.Result)
+	if err != nil {
+		logger.Error("Failed converting hex string to integer:", zap.Error(err))
+		return
+	}
+	ch <- height
+}
+
+func RunL2Analyzer(
+	zap *zap.Logger,
+	alerts chan<- entities.Alert,
+	nodeURL string,
+	nodeName string,
+) {
+	ch := make(chan int64)
+
+	go func() {
+		for {
+			collectL2Height(nodeURL, ch, zap)
+			time.Sleep(time.Minute)
```

Comment:
It's a bad way. Use a ticker for the task.
I also don't see a shutdown handling.
Also, `ch` should be close when the execution leaves this goroutine.